clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4684,6 +4684,25 @@ class SYCLKernelNameTypePrinter
   }
 };
 
+static void OutputStableNameChar(raw_ostream &O, char C) {
+  // If it is reliably printable, print in the integration header as a
+  // character. Else just print it as the integral representation.
+  if (C >= ' ' && C <= '~')
+    O << '\'' << C << '\'';
+  else
+    O << static_cast<short>(C);
```

Comment:
Do we want to print the values as being signed? I would have assumed we'd want unsigned char values (which I think we'd get if we cast to `uint8_t`).

Reply:
I don't think so.  The template parameter is a 'char', so the specialization would fail with the error:

````error: non-type template argument evaluates to 128, which cannot be narrowed to type 'char' [-Wc++11-narrowing]````

Reply:
Oh, interesting! Thanks!